src/utils/flattenSuggestions.ts

```
@@ -49,4 +88,61 @@ const flattenSuggestions = <Extra extends Record<string, unknown> = Record<strin
   return flattened
 }
 
+export const flattenSuggestionRenderEntries = <
+  Extra extends Record<string, unknown> = Record<string, unknown>,
+>(
+  children: ReactNode,
+  suggestions: SuggestionsMap<Extra> | undefined
+): Array<FlattenedSuggestionRenderEntry<Extra>> => {
```

ESLint

[error] Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed.

(sonarjs/cognitive-complexity)